logstash-core/lib/logstash/pipeline.rb

```diff
@@ -12,20 +12,32 @@
 require "logstash/config/cpu_core_strategy"
 require "logstash/util/defaults_printer"
 require "logstash/shutdown_controller"
+require "logstash/util/wrapped_synchronous_queue"
+require "logstash/pipeline_reporter"
 
 module LogStash; class Pipeline
-  attr_reader :inputs, :filters, :outputs, :input_to_filter, :filter_to_output
+  attr_reader :inputs, :filters, :outputs, :worker_threads, :events_consumed, :events_filtered, :reporter, :pipeline_id
+
+  DEFAULT_SETTINGS = {
+    # We don't set pipeline workers defaults here, but rather in safe_pipeline_workers
```

Comment:
I'm just leaning toward excising the whole safe_worker  thing at this point, it just makes the code more complicated. I suppose it could be in scope for this PR, and needn't be separate in #4272